firstvoices/backend/tasks/import_job_tasks.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+import tablib
+
+from backend.models.import_jobs import ImportJobReport, ImportJobReportRow, RowStatus
+from backend.resources.dictionary import DictionaryEntryResource
+
+
+def execute_dry_run_import(import_job_instance):
+    # This function will be modified later with a flag
+    # to be used for both dry-run and actual import
+
+    logger = logging.getLogger(__name__)
+
+    # Signals to be enabled during actual run, and not dry-run
+    # After a batch has been successfully uploaded, we should run a
+    # re-index for the site
+
+    # Disconnecting search indexing signals
+    # disconnect_signals()
+    # logger.info("Disconnected all search index related signals")
+
+    resource = DictionaryEntryResource(site=import_job_instance.site)
+
+    table = tablib.Dataset().load(
+        import_job_instance.data.content.open().read().decode("utf-8-sig"), format="csv"
+    )
+    table.headers = [header.lower() for header in table.headers]
+
+    try:
+        result = resource.import_data(dataset=table, dry_run=True)
+    except Exception as e:
+        logger.error(e)
+
+    # Create an ImportJobReport for the run
+    report = ImportJobReport(
+        site=import_job_instance.site,
+        importjob=import_job_instance,
+        new_rows=result.totals["new"],
+        skipped_rows=result.totals["skip"],
+        error_rows=result.totals["error"],
+    )
+    report.save()
```

Comment:
Question: It seems possible for the user to enter data that would result in an update or deletion (e.g. if they include an ID field that matches an existing entry). Do we want to track the other possible row values in the report, or anywhere in this job? Or will we handle this possibility later?

Reply:
Yes we have plans to identify the invalid column name and value later and then put them in the `identifier_field` and `identifier_value` fields of this model. As of now, the errors are just returned as is, for each row, and we are not parsing them for MVP.
Please let me know if we want to do that for MVP and I can create tickets for that.

Reply:
@caforbes Are you seeing a way in the current API for someone to cause an edit or delete instead of only adding rows?

Reply:
I get it now :) We logged a ticket to enforce the import mode we are intended to use (create vs edit).